### ISSUE TITLE
JLD2 not installing under latest 0.7-DEV

### ISSUE THREAD

**Issue by StephenVavasis** [simonster/JLD2.jl#61]
I'm trying to port my code from 0.6.2 to 0.7-DEV (0-day old master) and so far I haven't been able to install either JLD or JLD2 (Ubuntu 17.10).  The installation of JLD2 is failing because the code is apparently not able to find libz.  Below is the error message.

```
[ Info: Precompiling module JLD2
┌ Warning: Deprecated syntax `type` at /home/vavasis/.julia/v0.7/Libz/src/lowlevel.jl:62.
│ Use `mutable struct` instead.
└ @ nothing lowlevel.jl:62
┌ Warning: `is_windows` is deprecated, use `Sys.iswindows` instead.
│   caller = top-level scope
└ @ Core :0
WARNING: importing deprecated binding Base.Void into Libz.
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /home/vavasis/.julia/v0.7/Libz/src/lowlevel.jl:62
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /home/vavasis/.julia/v0.7/Libz/src/lowlevel.jl:62
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /home/vavasis/.julia/v0.7/Libz/src/lowlevel.jl:62
WARNING: Base.Void is deprecated, use Nothing instead.
  likely near /home/vavasis/.julia/v0.7/Libz/src/lowlevel.jl:62
ERROR: LoadError: LoadError: error compiling version: could not load library "libz"
libz.so: cannot open shared object file: No such file or directory
Stacktrace:
 [1] top-level scope
 [2] include at ./boot.jl:295 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1060
 [4] include at ./sysimg.jl:27 [inlined]
 [5] include(::String) at /home/vavasis/.julia/v0.7/Libz/src/Libz.jl:3
 [6] top-level scope
 [7] include at ./boot.jl:295 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1060
 [9] include(::Module, ::String) at ./sysimg.jl:27
 [10] top-level scope
 [11] eval at ./boot.jl:298 [inlined]
 [12] top-level scope at ./<missing>:3
in expression starting at /home/vavasis/.julia/v0.7/Libz/src/lowlevel.jl:110
in expression starting at /home/vavasis/.julia/v0.7/Libz/src/Libz.jl:11
ERROR: LoadError: Failed to precompile Libz to /home/vavasis/.julia/lib/v0.7/Libz.ji.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] compilecache(::Base.PkgId) at ./loading.jl:1198
 [3] _require(::Base.PkgId) at ./loading.jl:967
 [4] require(::Module, ::Symbol) at ./loading.jl:867
 [5] include at ./boot.jl:295 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1060
 [7] include(::Module, ::String) at ./sysimg.jl:27
 [8] top-level scope
 [9] eval at ./boot.jl:298 [inlined]
 [10] top-level scope at ./<missing>:3
in expression starting at /home/vavasis/.julia/v0.7/JLD2/src/JLD2.jl:4
ERROR: LoadError: Failed to precompile JLD2 to /home/vavasis/.julia/lib/v0.7/JLD2.ji.
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] compilecache(::Base.PkgId) at ./loading.jl:1198
 [3] _require(::Base.PkgId) at ./loading.jl:996
 [4] require(::Module, ::Symbol) at ./loading.jl:867
 [5] include at ./boot.jl:295 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1060
 [7] include(::Module, ::String) at ./sysimg.jl:27
 [8] include(::String) at ./sysimg.jl:66
 [9] top-level scope
in expression starting at /home/vavasis/ownCloud/Documents/Katerina/cohesive/conic_jl/conic29.jl:8

julia> 
```

**Comment by ararslan** [comment]
This appears to be an issue with Libz, not JLD2.

**Comment by StephenVavasis** [comment]
OK, I opened an issue in libz.
https://github.com/BioJulia/Libz.jl/issues/59

**Comment by bicycle1885** [comment]
I will not actively maintain Libz.jl because I started developing [CodecZlib.jl](https://github.com/bicycle1885/CodecZlib.jl), which is an alternative of handling compressed data formats. Quickly skimming the code of JLD2.jl, I think it is not so heavy to switch to CodecZlib.jl.

**Comment by kafisatz** [comment]
In addition to the above, I currently get an error (dev.4690) that SimpleVector is not defined (in data.jl line 1027). What is the easiest way to fix this?

**Comment by ararslan** [comment]
@bicycle1885 We shouldn't adopt CodecZlib here yet because it uses BinaryBuilder, which doesn't support all of the platforms that Julia supports. (See for example https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/32 and https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/126.) I'd prefer not to proliferate the exclusion of supported platforms.

**Comment by ararslan** [comment]
https://github.com/bicycle1885/CodecZlib.jl/pull/15 will address the platform concerns.

**Comment by bicycle1885** [comment]
CodecZlib.jl v0.4.3 includes the build support from source (https://github.com/JuliaLang/METADATA.jl/pull/14023). So, I think you can switch the dependency whenever you're ready. Thanks again, @ararslan.

**Comment by ararslan** [comment]
Great, thanks! I think we should go ahead with the switch then. @bicycle1885, if you're familiar with the primary differences between Zlib and CodecZlib, would you be able to submit a PR? I'd be happy to review it.

**Comment by bicycle1885** [comment]
Sure 😃

**Comment by zahachtah** [comment]
Didn't want to start new issue, under 0.7 nighlies I get error:

```julia
julia> using JLD2
[ Info: Precompiling module JLD2
ERROR: LoadError: ArgumentError: Module Pkg not found in current path.
Run `Pkg.add("Pkg")` to install the Pkg package.
```
installed using Pkg3

known or something I made a mistake. need jld2 to ge my package running :-)

**Comment by ararslan** [comment]
This was fixed by #67. Remaining 0.7 issues are logged separately.
